lib/Slackbot_worker.js

```diff
@@ -86,10 +105,22 @@ module.exports = function(botkit, config) {
             bot.rtm = new Ws(res.url, null, {agent: agent});
             bot.msgcount = 1;
 
-            var pingIntervalId = null;
+            bot.rtm.on('pong', function (obj) {
+                lastPong = Date.now()
+                botkit.debug('PONG received');
+            })
+
             bot.rtm.on('open', function() {
 
                 pingIntervalId = setInterval(function() {
+                    if (lastPong && lastPong + 12000 < Date.now()) {
```

Comment:
BTW, you're right -- I was doing it wrong.  `bot.rtm.on()` was the right method - I was peeking inside the `ws` module, and tried doing `bot.rtm._receiver.onpong()` directly, which wasn't working.

Nice work!